### PR TITLE
feat(SR7): rate limitter

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction
Rate limiter introduction

## Resolution
* Introduction `express-rate-limit` package with custom configuration.

### CODE REVIEW

File: portal/server/routes/middleware/rateLimit/index.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  if (!RATE_LIMIT_THRESHOLD) {
+    console.error('Invalid rate limit threshold value %d', RATE_LIMIT_THRESHOLD);
+    return res.redirect('/not-found');
+  }
+
+  return limitter({
+    // 1 minutes
+    windowMs: 1 * 60 * 1000,
+    // Threshold limit, x requests per windowMs (60 seconds)
+    max: RATE_LIMIT_THRESHOLD,
```
Comment (AlexBramhill) [inline portal/server/routes/middleware/rateLimit/index.js:20]: Were we not using a unique one for each service or just dtfs shared?

**Review by AlexBramhill — APPROVED** [review]

File: portal/server/routes/middleware/rateLimit/index.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const limitter = require('express-rate-limit');
+
+const { RATE_LIMIT_THRESHOLD } = process.env;
+
+/**
+ * Global rate-limitter middleware
```
Comment (AlexBramhill) [inline portal/server/routes/middleware/rateLimit/index.js:6]: limiter?

File: portal/server/routes/middleware/rateLimit/index.js (hunk 1 above)
Comment (abhi-markan) [reply portal/server/routes/middleware/rateLimit/index.js:20]: The name will be common, but values for each `.env` inside their respective micro-services can be different.

File: portal/server/routes/middleware/rateLimit/index.js (hunk 2 above)
Comment (AlexBramhill) [inline portal/server/routes/middleware/rateLimit/index.js:6]:
```suggestion
 * Global rate-limiter middleware
```

File: portal/server/routes/middleware/rateLimit/index.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * @param {Object} req Request object
+ * @param {Object} res Response object
+ */
+const rateLimit = (req, res) => {
+  if (!RATE_LIMIT_THRESHOLD) {
+    console.error('Invalid rate limit threshold value %d', RATE_LIMIT_THRESHOLD);
+    return res.redirect('/not-found');
+  }
+
+  return limitter({
```
Comment (AlexBramhill) [inline portal/server/routes/middleware/rateLimit/index.js:16]:
```suggestion
  return limiter({
```

File: portal/server/routes/middleware/rateLimit/index.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    // 1 minutes
+    windowMs: 1 * 60 * 1000,
+    // Threshold limit, x requests per windowMs (60 seconds)
+    max: RATE_LIMIT_THRESHOLD,
+    // Return rate limit info in the `RateLimit-*` headers
+    standardHeaders: true,
+    // Disable the `X-RateLimit-*` headers
+    legacyHeaders: true,
```
Comment (oscar-richardson-softwire) [inline portal/server/routes/middleware/rateLimit/index.js:24]: Should this value not be false to disable the legacyHeaders?

**Review by AlexBramhill — DISMISSED** [review]

**Review by IgnatG — CHANGES_REQUESTED** [review]
The rate limiter needs to be at API level. It's currently set in the UI

**Comment by lmarrai-sw** [comment]
@abhi-markan we can close this PR now since it is superseded by #1941

**Comment by abhi-markan** [comment]
PR closed as this PR has been superseded by #1941

**Pull request closed without merging**